content/en/docs/contribute/localization.md

```diff
@@ -128,7 +128,7 @@ Localizations must use English files from the most recent release as their sourc
 To find source files for the most recent release:
 
 1. Navigate to the Kubernetes website repository at https://github.com/kubernetes/website.
-2. Select the `release-1.X` branch for the most recent version.
+2. Select the most recent localization development branch or open it. Usually a team member of [@kubernetes/sig-docs-l10n-admins] opens the development branch from `master` branch on https://github.com/kubernetes/website. The `master` branch will be merged onto the most recent release branch eventually. See the [branching strategy](#branching-strategy) for more information.
```

Comment:
@gochist This section deals with finding source files in English, so it doesn't make sense to add instructions here about opening a development branch. I think your changes on line 172 👇 make a better place for this sort of expansion.